includes/class-wc-admin-api-init.php

```diff
@@ -315,6 +321,101 @@ public static function order_product_lookup_store_init( $updater = null ) {
 		return true;
 	}
 
+	/**
+	 * Init customers report data store.
+	 */
+	public static function customers_report_data_store_init() {
+		WC_Admin_Reports_Customers_Data_Store::init();
+	}
+
+	/**
+	 * Init customer lookup store.
+	 *
+	 * @param WC_Background_Updater|null $updater Updater instance.
+	 * @return bool
+	 */
+	public static function customer_lookup_store_init( $updater = null ) {
+		// TODO: this needs to be updated a bit, as it no longer runs as a part of WC_Install, there is no bg updater.
+		global $wpdb;
+
+		// Backfill customer lookup table with registered customers.
+		$customer_ids = get_transient( 'wc_update_350_all_customers' );
+
+		if ( false === $customer_ids ) {
+			$customer_query = new WP_User_Query(
+				array(
+					'fields' => 'ID',
+					'role'   => 'customer',
+					'number' => -1,
+				)
+			);
+
+			$customer_ids = $customer_query->get_results();
+
+			set_transient( 'wc_update_350_all_customers', $customer_ids, DAY_IN_SECONDS );
+		}
+
+		// Process customers until close to running out of memory timeouts on large sites then requeue.
+		foreach ( $customer_ids as $customer_id ) {
+			$result = WC_Admin_Reports_Customers_Data_Store::update_registered_customer( $customer_id );
+
+			if ( $result ) {
+				// Pop the customer ID from the array for updating the transient later should we near memory exhaustion.
+				unset( $customer_ids[ $customer_id ] );
+			}
+
+			if ( $updater instanceof WC_Background_Updater && $updater->is_memory_exceeded() ) {
+				// Update the transient for the next run to avoid processing the same orders again.
+				set_transient( 'wc_update_350_all_customers', $customer_ids, DAY_IN_SECONDS );
+				return true;
+			}
+		}
+
+		// Backfill customer lookup table with guests.
+		$guest_order_ids = get_transient( 'wc_update_350_all_guest_orders' );
+
+		if ( false === $guest_order_ids ) {
+			$guest_order_ids = wc_get_orders(
+				// TODO: restrict to certain order status?
```

Comment:
I think for now it's safe to store all of them as these will still provide insight into user interaction with the site and won't alter the actual order stats.

If we're connecting the order lookup table here, filtering by number of orders >=1 will only show customers with accepted order statuses.